src/main/java/org/embulk/input/zendesk/ZendeskInputPlugin.java

```diff
@@ -199,6 +200,17 @@ public void cleanup(final TaskSource taskSource, final Schema schema, final int
     public TaskReport run(final TaskSource taskSource, final Schema schema, final int taskIndex, final PageOutput output)
     {
         final PluginTask task = taskSource.loadTask(PluginTask.class);
+
+        if (Exec.isPreview()) {
+            checkValidTimeRangeThrowException(task);
+        }
+        else {
+            if (!isValidTimeRange(task)) {
```

Comment:
I'm wondering should we bring the check here to `resume` method instead of? The reason why I asked that because in non-incremental mode, threads will be spawn with number of pages accordingly, in the worst case if there are many threads are spawn but then die quickly due to `end_date` issue.

Another question, I see there is another scenario where user would like to import their data in specific time range with non-incremental target as well as non-incremental mode, it is not possible with current implementation, should we support it as well?